lib/srv/db/access_test.go

```diff
@@ -333,6 +343,26 @@ func TestAccessMySQL(t *testing.T) {
 			require.NoError(t, err)
 		})
 	}
+
+	t.Run("server version update on connection", func(t *testing.T) {
+		// Confirm the server version configured in the spec.
+		db, err := testCtx.server.getProxiedDatabase("version-update-test")
+		require.NoError(t, err)
+		require.Equal(t, "6.6.6-before", db.GetMySQLServerVersion())
+
+		// Connect.
+		testCtx.createUserAndRole(ctx, t, "alice", "admin", []string{"alice"}, []string{types.Wildcard})
+		mysqlConn, err := testCtx.mysqlClient("alice", "version-update-test", "alice")
+		require.NoError(t, err)
+		defer mysqlConn.Close()
+		_, err = mysqlConn.Execute("select 1")
+		require.NoError(t, err)
+
+		// Check if proxied database is updated.
+		updatedDB, err := testCtx.server.getProxiedDatabase("version-update-test")
+		require.NoError(t, err)
+		require.Equal(t, "8.8.8-after", updatedDB.GetMySQLServerVersion())
+	})
```

Comment:
nit: 
Can you move this into a sperate  function test like `TestServerVersionUpgradeOnConnection(t *testing.t)` 

Currently the TestAccessMySQL  test case is  using table tests approach and adding  "server version update on connection" after table tests is a bit unreadable.

Reply:
thanks! moved!